pyxem/utils/plotting_utils.py

```diff
@@ -113,3 +118,187 @@ def plot_template_over_pattern(
         color=marker_color,
     )
     return (ax, im, sp)
+
+
+def generate_template_markers(
+    signal,
+    library,
+    result: dict,
+    phase_key_dict: dict,
+    n_best: int = None,
+    direct_beam_position: tuple[int, int] = None,
+    marker_colors: list[str] = None,
+    scale_markers: bool = True,
+    size_factor: float = 1.0,
+) -> Iterator[hs.plot.markers.Markers]:
+    """
+    Generate markers of simulated diffraction patterns corresponding to template matching results.
+    These can be added to a signal plot for template matching result inspection.
+
+    Parameters
+    ----------
+    signal : hyperspy.signals.Signal2D
+        The 4D-STEM dataset.
+    library : diffsims.libraries.diffraction_library.DiffractionLibrary
+        The library of simulated diffraction patterns.
+    result : dict
+        Template matching results dictionary containing keys: phase_index, template_index,
+        orientation, correlation, and mirrored_template.
+        Returned from pyxem.utils.indexation_utils.index_dataset_with_template_rotation.
+    phase_key_dict: dict
+        A small dictionary to translate the integers in the phase_index array
+        to phase names in the original template library.
```

Comment:
I'd suggest only having the type in one place.  Either in the function arguments or in the doc strings otherwise you can run into the problem of the two of them not agreeing and that becomes a pain to manage.

Reply:
Sorry to be a pain here, but I really think types should be included in the docstrings.

Reply:
@pc494 maybe this deserves more discussion. I used to feel the same way but I think you really don't want two different types as sometimes things like IDE's will use both types. 

For example `orix` which is much more completely typed only has the types in the doc strings. (side note it would be nice to have the types in the docstring). I'm pretty sure that could just be done with https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_typehints

The nicest thing about typing(that pyxem doesn't do well) is that it gets the intersphinx mapping correct. Honestly I think that this is the nicest thing about typing.

@hakonanes Any thoughts?